pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -411,8 +412,8 @@ private CompletableFuture<Optional<LookupResult>> findBrokerServiceUrl(
                         });
                     }
                 } else if (nsData.get().isDisabled()) {
-                    future.completeExceptionally(
-                            new IllegalStateException(String.format("Namespace bundle %s is being unloaded", bundle)));
+                    future.completeExceptionally(new BrokerServiceException.ServerMetadataException(
```

Comment:
Is there only one way that “ bundle is being unloaded” can cause `nsData.get().isDisabled()`?  If so, I think it's a good change.

Reply:
I think so. A bundle is supposed to belong to a broker.